core/src/test/scala/org/apache/spark/serializer/UnsafeKryoSerializerSuite.scala

```diff
@@ -32,4 +34,19 @@ class UnsafeKryoSerializerSuite extends KryoSerializerSuite {
     conf.set(KRYO_USE_UNSAFE, false)
     super.afterAll()
   }
+
+  test("SPARK-27216: Upgrade RoaringBitmap to 0.7.45 to fix Kryo unsafe ser/dser issue") {
```

Comment:
Please move your test into `KryoSerializerSuite` where `conf.set(KRYO_USE_UNSAFE, false)` is given. 
And as `UnsafeKryoSerializerSuite` is inherited from `KryoSerializerSuite` beside using `conf.set(KRYO_USE_UNSAFE, true)` in `beforeAll()` it will **inherit your test method** so both the safe/unsafe case will be tested by this. 

Summary: you do not need to test both cases in one test method.

Reply:
Oh, thanks for this reminding.